pkg/runtime.v2/framework/plugins/plainml/plainml.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plainml
+
+import (
+	"context"
+
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtime "github.com/kubeflow/training-operator/pkg/runtime.v2"
+	"github.com/kubeflow/training-operator/pkg/runtime.v2/framework"
+)
+
+var _ framework.EnforceMLPolicyPlugin = (*PlainML)(nil)
+
+type PlainML struct{}
+
+const Name = "PlainML"
+
+func New(context.Context, client.Client, client.FieldIndexer) (framework.Plugin, error) {
+	return &PlainML{}, nil
+}
+
+func (p *PlainML) Name() string {
+	return Name
+}
+
+func (p *PlainML) EnforceMLPolicy(info *runtime.Info) error {
+	if info == nil || info.MLPolicy == nil || info.MLPolicy.Torch != nil || info.MLPolicy.MPI != nil {
+		return nil
+	}
+	numNodes := ptr.Deref(info.MLPolicy.NumNodes, 1)
+	for rName := range info.TotalRequests {
+		info.TotalRequests[rName] = runtime.TotalResourceRequest{
+			Replicas:    numNodes,
+			PodRequests: info.TotalRequests[rName].PodRequests,
+		}
+	}
```

Comment:
How are we using this while enforcing the MLPolicy ?

Reply:
```
if info == nil || info.MLPolicy != nil {
```

I wanted to implement this in line 43. Maybe I failed to reabase.
Let me fix this.

Reply:
NVM above comments.

Reply:
```yaml
spec:
  mlPolicy:
    numNodes: 1
```

We can imagine this situation.

Reply:
I see, so we override the value that we set here, right ?
https://github.com/kubeflow/training-operator/blob/0c376d3425c291f43da8c300e0742aa117390936/pkg/runtime.v2/runtime.go#L120-L125

Reply:
Yes, this plainml codes try to update with the proper one.